pkgs/http/lib/src/client.dart

```diff
@@ -139,7 +140,11 @@ abstract class Client {
   Future<Uint8List> readBytes(Uri url, {Map<String, String>? headers});
 
   /// Sends an HTTP request and asynchronously returns the response.
-  Future<StreamedResponse> send(BaseRequest request);
+  ///
+  /// If [onUploadProgress] defined, the callback will be called when the
+  /// upload progress changes.
+  Future<StreamedResponse> send(BaseRequest request,
```

Comment:
Some drive-by comments:
1. this change is clearly breaking (and would break existing code in practice) and should be paired with a significant version bump
2. IIUC, this is fairly easy functionality for the user to replicate
3. This functionality seems generally useful

Maybe there is another way of providing this functionality? Like set the callback in the constructor? Or as a property?

Reply:
I just needed to add the callback as argument in MultipartRequest's constructor.  But it will be handled by implementations of `Client.send`.  If you meant the `Client` constructor, I think it's a bad idea.  Because OnUploadProgress's context is send, not `Client`.

Reply:
Also, adding this feature will only give an error feom old codes if the developer has written developer's own `Client` implementation.

In external cases, there will be no errors or conflicts  because it is a nullable named argument.

Reply:
I understand. 

I think we can resolve these conflicts with a nullable property to add to `BaseRequest`.  `send` is already take a `BaseRequest`.  I didn't get a chance to try with coding.  I will try as soon as possible.

I think the property you mentioned was for `BaseRequest`.  I thought it wouldn't be a solution because I thought you said it for the `Client`.

Reply:
I remove the argument from `Client.send`. It's a really useless argument. Because the only place that works `send` with `onUploadProgress` is `BaseRequest` and `send` already gets `BaseRequest`.